app/src/main/java/org/mozilla/FirefoxLite/firefoxdownloader/MainActivity.java

```diff
@@ -0,0 +1,294 @@
+
+            if (statusofdownload!=null){
+                if(statusofdownload.equals("started")){
+                    progressBarOne.setIndeterminate(false);
+                    buttonOne.setEnabled(true);
+                    buttonOne.setText(R.string.pause);
+                    buttonCancelOne.setEnabled(true);
+                    buttonScheduleOne.setEnabled(false);
+                    progressBarOne.setProgress(0);
+                    textViewProgressOne.setText("");
+                }
+                else if (statusofdownload.equals("paused")){
+                    buttonOne.setText(R.string.resume);
+                }
+
+                else if (statusofdownload.equals("cancelled")){
+                    buttonOne.setText(R.string.start);
+                    buttonCancelOne.setEnabled(false);
+                    progressBarOne.setProgress(0);
+                    textViewProgressOne.setText("");
+                    progressBarOne.setIndeterminate(false);
+                    schedulesession="false";
+                    buttonScheduleOne.setEnabled(true);
+                }
+
+            }
+        }
+
+    }
+    public void startService(View v){
+        Intent serviceintent=new Intent(this,SampleService.class);
+        startService(serviceintent);
+
+    }
+    @Override
+    protected void onStop() {
+        unregisterReceiver(myReceiver);
+        super.onStop();
+    }
+    public void stopService(View v){
+        Intent serviceintent=new Intent(this,SampleService.class);
+        stopService(serviceintent);
+    }
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        myReceiver = new MyReceiver();
+        IntentFilter intentFilter = new IntentFilter();
+        intentFilter.addAction(SampleService.MY_ACTION);
+        registerReceiver(myReceiver, intentFilter);
+        mNotificationHelper=new NotificationHelper(this);
+        mainActivity = this;
+        lasttimestamp=System.currentTimeMillis();
+        dirPath = Utils.getRootDirPath(getApplicationContext());
+        /**Getting the external storage permissions**/
+        if (ContextCompat.checkSelfPermission(MainActivity.this, Manifest.permission.READ_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED)
+        {
+            Toast.makeText(this, "You have already been granted the permissions", Toast.LENGTH_SHORT).show();
+        }
+        else
+        {
+            requeststoragepermission();
+        }
+        //Code to autostart
+        //        if (Build.BRAND.equalsIgnoreCase("xiaomi")) {
+        //            Intent intent = new Intent();
+        //            intent.setComponent(new ComponentName("com.miui.securitycenter", "com.miui.permcenter.autostart.AutoStartManagementActivity"));
+        //            startActivity(intent);
+        //        }
+        init();
+        onClickListenerOne();
```

Comment:
In your demo app, you should separate the logic of creating a download, and displaying a download.
Now your download progress will only be visible if it present before it's downloaded. This is impossible in real-world cause you can only know the download status after you start the download.